core/src/main/java/org/ehcache/core/Ehcache.java

```diff
@@ -183,30 +190,19 @@ public void compute(K key, final BiFunction<? super K, ? super V, ? extends V> c
             getObserver.end(GetOutcome.HIT);
           }
 
-          V newValue = computeFunction.apply(mappedKey, mappedValue);
+          return computeFunction.apply(mappedKey, mappedValue);
 
-          if (newValue == mappedValue) {
-            if (! replaceEqual.get()) {
-              return mappedValue;
-            }
-          }
-
-          if (newValueAlreadyExpired(mappedKey, mappedValue, newValue)) {
```

Comment:
This `newValueAlreadyExpired` check was intentionally dropped?

Reply:
yes